### PR TITLE
WIP: prebuild wasm binaries

### PR DESCRIPTION
**Pull request by TriplEight**
Optimizing CI by prebuilding wasm binaries and distributing it between with artifacts

### CODE REVIEW

**Comment by bkchr** [comment]
They would be rebuild for every pr?

**Comment by TriplEight** [comment]
Yeah, since test-linux requires them

**Comment by TriplEight** [comment]
@bkchr can you please tell me exactly what files are needed there?

**Comment by bkchr** [comment]
`core/test-runtime/wasm/target/wasm32-unknown-unknown/release/substrate_test_runtime.compact.wasm`
`node/runtime/wasm/target/wasm32-unknown-unknown/release/node_runtime.compact.wasm`
`core/executor/wasm/target/wasm32-unknown-unknown/release/runtime_test.compact.wasm`

This should be all the relevant ones.

**Comment by TriplEight** [comment]
@bkchr this is how it looks right now: https://gitlab.parity.io/parity/substrate/-/jobs/161095
this means all the `*.wasm` files will be available as `/artifacts/*.wasm` in the next job which summons them (build/test/etc job).
Should only those files without their surroundings in `target` dirs be enough?

**Comment by TriplEight** [comment]
Is it okay if I'll change all the occurrences of these files in the code to `./artifacts/.../*.wasm`?
I mean this would be totally OK with GitLab CI way to build Substrate. 
But is there anyone who builds it other way?

**Comment by TriplEight** [comment]
Alternatively, I could `cp` these `*.wasm` files to some dir into `./scripts/build.sh`

File: scripts/build.sh [hunk 1]
```diff
@@ -4,25 +4,25 @@
 
 set -e
 
-PROJECT_ROOT=`git rev-parse --show-toplevel`
-source "`dirname \"$0\"`/common.sh"
+PROJECT_ROOT=`git rev-parse --show-toplevel`    # /home/tripleight/code/github.com/paritytech/substrate
+source "`dirname \"$0\"`/common.sh"             # source scripts/common.sh
```
Comment (DemiMarie-parity) [inline scripts/build.sh]: This is wrong if `dirname "$0"` ends with a newline, but I don’t think we need to worry about this.

File: scripts/build.sh [hunk 2]
```diff
@@ -4,25 +4,25 @@
...
 
-for SRC in "${SRCS[@]}"
+for SRC in "${SRCS[@]}"                         # SRCS=("core/executor/wasm" "node/runtime/wasm" "node-template/runtime/wasm"	"core/test-runtime/wasm")
 do
   echo "*** Building wasm binaries in $SRC"
-  cd "$PROJECT_ROOT/$SRC"
-
-  ./build.sh
+  cd "$PROJECT_ROOT/$SRC"                       # 
+
```
Comment (DemiMarie-parity) [inline scripts/build.sh]: Trailing whitespace
```suggestion

```

**Comment by DemiMarie-parity** [comment]
> Alternatively, I could `cp` these `*.wasm` files to some dir in `./scripts/build.sh`

This really needs to be done in `cargo build`.

File: scripts/build.sh (hunk 1 above)
Comment (TriplEight) [reply scripts/build.sh]: I'm about to simplify this.

**Comment by TriplEight** [comment]
it's still WIP

**Comment by bkchr** [comment]
I think we can close this for now? If not, please reopen.

**Comment by TriplEight** [comment]
Yes, thanks, now it's not needed.

**Pull request closed without merging**